docs/guide/cursors.txt

```diff
@@ -1,3 +1,164 @@
 =======
 Cursors
 =======
+
+When a driver executes a read that returns mutliple documents, the server does not immediately return all values that match the query. Instead, the driver creates a **Cursor** object which fetches the documents in batches.
+
+Where are cursors used in the driver?
+-------------------------------------
+
+The following functions directly return cursors to the user:
+
+
+* ``Collection.prototype.find``
+* ``Collection.prototype.aggregate``
+* ``Collection.prototype.listIndexes``
+* ``Db.prototype.aggregate``
+* ``Db.prototype.listCollections``
+
+In addition, many other methods, like ``Collection.prototype.findOne`` and ``Collection.prototype.watch``\ , use cursors in order to return results.
+
+How do I use a cursor?
+----------------------
+
+There are multiple ways to consume a cursor:
+
+Stream API
+^^^^^^^^^^
+
+All cursors in the Node Driver are Node Readable Streams operating in Object Mode. Cursors will work with most Node stream APIs.
+
+.. code-block:: js
+
+   // Using the stream API
+   const cursor = collection.find({});
+   cursor.pipe(new stream.Writable({
```

Comment:
why not just `cursor.pipe(process.stdout);`?

Reply:
I was trying to give an example of how you could specifically get the individual documents. But I guess that is a little overkill. Maybe we should provide two examples: one to specifically pipe directly to `stdout`, another to use a `TransformStream` to pipe to `stdout`, and a third to use a `Transform` to stringify, and then pipe to a `Writable` stream for a file?

Reply:
Also worth noting that piping directly into `stdout` is not the same as calling `console.log`